tools/blocksconvert/builder/series_iterator.go

```diff
@@ -0,0 +1,165 @@
+package builder
+
+import (
+	"encoding/gob"
+	"io"
+	"os"
+
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/tsdb/errors"
+)
+
+type seriesIterator struct {
+	files []*seriesFile
+	errs  errors.MultiError
+}
+
+func newSeriesIterator(files []*seriesFile) *seriesIterator {
+	si := &seriesIterator{
+		files: files,
+	}
+	si.buildHeap()
+	return si
+}
+
+func (sit *seriesIterator) buildHeap() {
+	for ix := 0; ix < len(sit.files); {
```

Comment:
I would add a comment that this `for` loop is used to remove files from which there's nothing more to read.